middleware/proxy/proxy.go

```diff
@@ -171,6 +171,39 @@ func Do(c *fiber.Ctx, addr string, clients ...*fasthttp.Client) error {
 	return nil
 }
 
+func DoRedirects(c *fiber.Ctx, addr string, maxRedirectsCount int, clients ...*fasthttp.Client) error {
+	var cli *fasthttp.Client
+	if len(clients) != 0 {
+		// Set local client
+		cli = clients[0]
+	} else {
+		// Set global client
+		lock.RLock()
+		cli = client
+		lock.RUnlock()
+	}
+	req := c.Request()
+	res := c.Response()
+	originalURL := utils.CopyString(c.OriginalURL())
+	defer req.SetRequestURI(originalURL)
+
+	copiedURL := utils.CopyString(addr)
+	req.SetRequestURI(copiedURL)
+	// NOTE: if req.isTLS is true, SetRequestURI keeps the scheme as https.
+	// issue reference:
+	// https://github.com/gofiber/fiber/issues/1762
+	if scheme := getScheme(utils.UnsafeBytes(copiedURL)); len(scheme) > 0 {
+		req.URI().SetSchemeBytes(scheme)
+	}
+
+	req.Header.Del(fiber.HeaderConnection)
+	if err := cli.DoRedirects(req, res, maxRedirectsCount); err != nil {
```

Comment:
```go

// Do performs the given http request and fills the given http response.
// This method can be used within a fiber.Handler
func Do(c *fiber.Ctx, addr string, clients ...*fasthttp.Client) error {
	var client *fasthttp.Client
	if len(clients) != 0 {
		// Set local client
		client = clients[0]
	} else {
		// Set global client
		lock.RLock()
		client = client
		lock.RUnlock()
	}
	
	return do(c, addr, func(cli *fasthttp.Client, req *fasthttp.Request, resp *fasthttp.Response, params ...interface{}) error {
		return cli.Do(req, resp)
	}, client)
}

func do(
	c *fiber.Ctx,
	addr string,
	action func(cli *fasthttp.Client, req *fasthttp.Request, resp *fasthttp.Response, params ...interface{}) error,
	client *fasthttp.Client,
	params ...interface{},
) error {
	req := c.Request()
	res := c.Response()
	originalURL := utils.CopyString(c.OriginalURL())
	defer req.SetRequestURI(originalURL)

	copiedURL := utils.CopyString(addr)
	req.SetRequestURI(copiedURL)
	// NOTE: if req.isTLS is true, SetRequestURI keeps the scheme as https.
	// issue reference:
	// https://github.com/gofiber/fiber/issues/1762
	if scheme := getScheme(utils.UnsafeBytes(copiedURL)); len(scheme) > 0 {
		req.URI().SetSchemeBytes(scheme)
	}

	req.Header.Del(fiber.HeaderConnection)
	if err := action(client, req, res, params...); err != nil {
		return err
	}
	res.Header.Del(fiber.HeaderConnection)
	return nil
}
```
something like this, but better

Reply:
@ReneWerner87 Do we want to add `DoTimeout` and `DoDeadline` to this PR?

Reply:
@ReneWerner87 Implementef the changes for DoTimeout and DoDeadlind. Having a hard time writing the unit-tests. Seems like I need to reuse the `createProxyServer` since `app.Test()` is returning 404 unless I use a real domain name.